lib/charms/opensearch/v0/opensearch_plugin_manager.py

```diff
@@ -50,399 +37,218 @@
-    ) -> Tuple[Dict[str, str], Dict[str, str]]:
-        """Returns the current and the new configuration."""
-        if not self._charm.opensearch.is_node_up() or not self.is_ready_for_api():
-            return None, None
-
-        current_settings = self.cluster_config
-        # We use current_settings and new_conf and check for any differences
-        # therefore, we need to make a deepcopy here before editing new_conf
-        # Also, we simply apply the config.config_entries straight to new_conf
-        # As setting a config entry to None will render a "null" entry with
-        # jsonl dumps, and hence, it will be removed from the configuration.
-        new_conf = copy.deepcopy(current_settings) | config.config_entries
-
-        logger.debug(
-            "Difference between current and new configuration: \n"
-            + str(
-                {
-                    "in current but not in new_conf": {
-                        k: v for k, v in current_settings.items() if k not in new_conf.keys()
-                    },
-                    "in new_conf but not in current": {
-                        k: v for k, v in new_conf.items() if k not in current_settings.keys()
-                    },
-                    "in both but different values": {
-                        k: v
-                        for k, v in new_conf.items()
-                        if k in current_settings.keys() and current_settings[k] != v
-                    },
-                }
+    relation_name = SMTP_RELATION
+    secret_label = SMTP_SECRET_LABEL
+
+    def __init__(self, charm: "OpenSearchBaseCharm"):
+        super().__init__(charm, "plugin:notifications")
+        self.charm = charm
+        self.smtp = SmtpRequires(self.charm, self.relation_name)
+
+        self.framework.observe(self.smtp.on.smtp_data_available, self._on_smtp_credentials_changed)
+        self.framework.observe(self.charm.on.smtp_relation_broken, self._on_smtp_credentials_gone)
+        self.framework.observe(self.charm.on.secret_changed, self._on_secret_changed)
+
+    def _on_smtp_credentials_changed(self, event) -> None:
+        """Creates secret containing key, value pairs for keystore"""
+        if not self.charm.opensearch.is_started():
+            # node must be reachable to reload settings after adding keys
+            event.defer()
+            return
+
+        # return if no relation data
+        if not (parameters := self.smtp.get_relation_data()):
+            logger.debug("Missing relation data from %s", self.relation_name)
+            self.charm.status.set(BlockedStatus(SmtpNoRelationData))
+            return
+
+        self.charm.status.clear(SmtpNoRelationData)
+
+        # validate data
+        required_params = ["user", "password"]
+        if missing_parameters := [p for p in required_params if not getattr(parameters, p)]:
+            logger.error("Parameters missing from smtp-intgrator: %s" % missing_parameters)
```

Comment:
There is a typo in integrator and missing_parameters is a  list, we can first create string to make it prettier.
```suggestion
             logger.error("Parameters missing from smtp-integrator: %s", ", ".join(missing_parameters))
```